llvm-spirv/lib/SPIRV/OCLUtil.cpp

```diff
@@ -531,6 +531,23 @@ class OCLBuiltinFuncMangleInfo : public SPIRV::BuiltinFuncMangleInfo {
       addVoidPtrArg(1);
       addUnsignedArg(2);
       addUnsignedArg(3);
+      // TODO: REMOVE BEFORE COMMIT!!!
```

Comment:
I think you left in an excess comment by accident!

Reply:
No, it is done by purpose :)
Actually I added this comment just before I push the patch to review. I will remove it after review is completed.

Reply:
Ah I apologize then!

Reply:
Please consider this comment as a proposal. I would really like to see OpenCL-like looking LLVM IR in the translator's output (like __read_pipe_bl_2), because it's easier to work with in middle-end. But I understand that this way is quiet incorrect, since it creates difference between LLVM IR before the translator and after.

Reply:
From the perspective of the current Xilinx implementation it's fine on our end, we currently don't make use of the SPIRV translator, we just take the initial LLVM bitcode passed to the ToolChain and do some minor transformations to make our backend happy. As long as there is no enforcement to use the translator I don't see a problem.

Although as a personal opinion, I'm fine with generating some kind of intermediate IR (OpenCL looking or otherwise) that can be easily redirected/source-to-source translated by an LLVM pass or middle-end. I think it may just have to be documented reasonably somewhere stating that X will change to Y (I believe you have already done at least some of this). However, maybe @keryell has another stronger opinion on the matter.

Reply:
Thanks for sharing your opinion!